src/main/java/ws/wamp/jawampa/WampClient.java

```diff
@@ -196,9 +199,9 @@ public RegisteredProceduresMapEntry(Subscriber<? super Request> subscriber, Regi
     
     HashMap<Long, RequestMapEntry> requestMap = 
         new HashMap<Long, WampClient.RequestMapEntry>();
-    
-    HashMap<String, SubscriptionMapEntry> subscriptionsByUri =
-        new HashMap<String, SubscriptionMapEntry>();
+
+    EnumMap<SubscriptionFlags, HashMap<String, SubscriptionMapEntry>> subscriptionsByFlags =
```

Comment:
Is there any reason why we need subscriptionByFlags here? I guess for cases where the client is subscribed to the same topic with both a prefix and an exact subscription?